ccl/logictestccl: TestCCLLogic failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/4c52530a33367c58e434111324fcda8c3d73582a

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1568679&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.81s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic875843548
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (0.61s)
            logic.go:1896: 
                
                testdata/logic_test/role:195: 
                expected success, but found
                (42501) testuser is not a superuser or role admin for role testrole
                role.go:97: in func1()
            logic.go:2299: 
                testdata/logic_test/role:195: error while processing
            logic.go:2299: pq: testuser is not a superuser or role admin for role testrole
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/320c1aad52af8a31afd1d5ef86a95e91aac7a446

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1570487&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.63s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic271663015
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
rewrote: CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (1), PARTITION p1 VALUES IN (2));
rewrote: CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a) (PARTITION p1 VALUES FROM (1) TO (2), PARTITION p1 VALUES FROM (2) TO (3));
rewrote: CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (1), PARTITION p1 VALUES IN (2));
rewrote: CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (1) PARTITION BY LIST (b) (PARTITION p1 VALUES IN (2)));
rewrote: CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (1) PARTITION BY RANGE (b) (PARTITION p1 VALUES FROM (1) TO (2)));
rewrote: CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a) (PARTITION p1 VALUES FROM (0) TO (1) PARTITION BY LIST (b) (PARTITION p2 VALUES IN (2)));
--- done: testdata/logic_test/case_sensitive_names with config local: 2 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (0.66s)
            logic.go:1658: 
                 
                testdata/logic_test/role:198: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  testrole   YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2299: 
                testdata/logic_test/role:204: error while processing
            logic.go:2299: testdata/logic_test/role:204: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/b9dc16b9d6cbd4627bac4b6329e87e29e52401b9

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1571344&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.20s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic342500896
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
--- done: testdata/logic_test/case_sensitive_names with config local: 2 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (1.32s)
            logic.go:1658: 
                 
                testdata/logic_test/role:357: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2299: 
                testdata/logic_test/role:363: error while processing
            logic.go:2299: testdata/logic_test/role:363: too many errors encountered, skipping the rest of the input

```

SHA: https://github.com/cockroachdb/cockroach/commits/91bc9ec64f1321ca39fe9838fea584db455726ab

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1573300&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (8.10s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic072411634
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (2.09s)
            logic.go:1659: 
                 
                testdata/logic_test/role:372: SELECT * FROM information_schema.enabled_roles
                expected:
                    role_name
                    rolea
                    roleb
                    rolec
                    roled
                    testuser
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    role_name
                    testuser
                    
                
            logic.go:2300: 
                testdata/logic_test/role:382: error while processing
            logic.go:2300: testdata/logic_test/role:382: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/98e73d2466a6252a11e5652debf910da2a0afbb4

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1576154&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.51s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic634816832
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.30s)
            logic.go:1659: 
                 
                testdata/logic_test/role:357: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2300: 
                testdata/logic_test/role:363: error while processing
            logic.go:2300: testdata/logic_test/role:363: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/dcddfec777c9c799ed5717e8f14899cfdb03b6e6

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1578900&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.97s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic228068513
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
rewrote: CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));
rewrote: CREATE TABLE orders (region STRING NOT NULL, id INT8 NOT NULL, total DECIMAL NOT NULL, created_at TIMESTAMP NOT NULL, PRIMARY KEY (region, id), FAMILY (created_at), FAMILY (total));
--- done: testdata/logic_test/as_of with config local: 3 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (2.14s)
            logic.go:1897: 
                
                testdata/logic_test/role:493: 
                expected success, but found
                (42501) testuser is not a superuser or role admin for role rolea
                role.go:97: in func1()
            logic.go:2203: 
                testdata/logic_test/role:493: error while processing
            logic.go:2203: pq: testuser is not a superuser or role admin for role rolea
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/ae07b57a07428f3fb9b1d26d98b8204766091dad

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1581011&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (7.00s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic999417589
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
--- done: testdata/logic_test/as_of with config local: 3 tests, 0 failures
rewrote: CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));
        --- FAIL: TestCCLLogic/local/role (1.68s)
            logic.go:1659: 
                 
                testdata/logic_test/role:394: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2203: 
                testdata/logic_test/role:400: error while processing
            logic.go:2203: testdata/logic_test/role:400: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/33b96613ae532b25a1b6b716453bece9b60ba2d6

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1583819&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.78s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic121405864
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.52s)
            logic.go:1659: 
                 
                testdata/logic_test/role:394: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2203: 
                testdata/logic_test/role:400: error while processing
            logic.go:2203: testdata/logic_test/role:400: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/35e138aa3c2be545fb4e17a85ea6f1b8d6525e53

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1584830&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (7.70s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic612858461
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (2.23s)
            logic.go:1659: 
                 
                testdata/logic_test/role:357: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2203: 
                testdata/logic_test/role:363: error while processing
            logic.go:2203: testdata/logic_test/role:363: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/0e9dd73f803247cdcfd06f51ce6b23396af1b9f5

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1587188&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.87s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic052342782
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.39s)
            logic.go:1662: 
                 
                testdata/logic_test/role:400: SELECT * FROM information_schema.applicable_roles
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolec      NO
                    testuser  roleb      NO
                    testuser  rolea      YES
                    testuser  roled      NO
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2206: 
                testdata/logic_test/role:409: error while processing
            logic.go:2206: testdata/logic_test/role:409: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/07b5c74deee727e9e74705f4a989e0f6923d99d6

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1589163&tab=buildLog

```
                                                                          num_replicas = 5,
                                                                          constraints = '[]',
                                                                          lease_preferences = '[]'
                        RANGE system                                      ALTER RANGE system CONFIGURE ZONE USING
                                                                          range_min_bytes = 16777216,
                                                                          range_max_bytes = 67108864,
                                                                          gc.ttlseconds = 90000,
                                                                          num_replicas = 5,
                                                                          constraints = '[]',
                                                                          lease_preferences = '[]'
                        TABLE system.public.jobs                          ALTER TABLE system.public.jobs CONFIGURE ZONE USING
                                                                          range_min_bytes = 16777216,
                                                                          range_max_bytes = 67108864,
                                                                          gc.ttlseconds = 600,
                                                                          num_replicas = 5,
                                                                          constraints = '[]',
                                                                          lease_preferences = '[]'
                        TABLE system.public.replication_constraint_stats  ALTER TABLE system.public.replication_constraint_stats CONFIGURE ZONE USING
                                                                          gc.ttlseconds = 600,
                                                                          constraints = '[]',
                                                                          lease_preferences = '[]'
                        TABLE system.public.replication_stats             ALTER TABLE system.public.replication_stats CONFIGURE ZONE USING
                                                                          gc.ttlseconds = 600,
                                                                          constraints = '[]',
                                                                          lease_preferences = '[]'
                        TABLE test.public.t3                              ALTER TABLE test.public.t3 CONFIGURE ZONE USING
                                                                          num_replicas = 8,
                                                                          constraints = '[]',
                                                                          lease_preferences = '[]'
                        TABLE test.public.t4                              ALTER TABLE test.public.t4 CONFIGURE ZONE USING
                                                                          num_replicas = 7
                        
                    
                logic.go:1314: 
                     testdata/logic_test/crdb_internal:113: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local/crdb_internal
=== PAUSE TestCCLLogic/local/crdb_internal
=== CONT  TestCCLLogic/local/crdb_internal
rewrote: CREATE TABLE t1 (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c), INDEX (a, b) PARTITION BY LIST (a, b) (PARTITION p00 VALUES IN ((0, 0)))) PARTITION BY LIST (a) (PARTITION p12 VALUES IN (1, 2) PARTITION BY LIST (b) (PARTITION p12p3 VALUES IN (3) PARTITION BY LIST (c) (PARTITION p12p3p8 VALUES IN (8)), PARTITION pd VALUES IN (DEFAULT)), PARTITION p6 VALUES IN (6) PARTITION BY RANGE (b) (PARTITION p6p7 VALUES FROM (minvalue) TO (7), PARTITION p6p8 VALUES FROM (7) TO (8), PARTITION p6px VALUES FROM (8) TO (maxvalue)));
        --- FAIL: TestCCLLogic/local/crdb_internal (1.38s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/8622fad01478fb4a4f05b5579eb0b8561c02e491

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1591148&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.60s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic900834598
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
--- done: testdata/logic_test/restore with config local: 1 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (1.15s)
            logic.go:1662: 
                 
                testdata/logic_test/role:357: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2206: 
                testdata/logic_test/role:363: error while processing
            logic.go:2206: testdata/logic_test/role:363: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/f96d984ba0be3f4f4255e426aafa1e42e0a5f82f

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1592925&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.22s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic928507213
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.61s)
            logic.go:1662: 
                 
                testdata/logic_test/role:357: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2206: 
                testdata/logic_test/role:363: error while processing
            logic.go:2206: testdata/logic_test/role:363: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/e78a2b657574e3842cd8fd50ed92d0411a9bffef

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1594091&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.99s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic569718359
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (2.45s)
            logic.go:1900: 
                
                testdata/logic_test/role:615: SELECT * FROM system.role_members
                expected success, but found
                (42501) user testuser does not have SELECT privilege on relation role_members
                authorization.go:116: in CheckPrivilege()
            logic.go:1662: 
                 pq: user testuser does not have SELECT privilege on relation role_members
            logic.go:2206: 
                testdata/logic_test/role:623: error while processing
            logic.go:2206: testdata/logic_test/role:623: too many errors encountered, skipping the rest of the input

```

SHA: https://github.com/cockroachdb/cockroach/commits/09b391536f72349c3951ce6b75c5231b58933b07

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1594990&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (8.09s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic497870571
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
rewrote:
CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));


--- done: testdata/logic_test/restore with config local: 1 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (2.06s)
            logic.go:1662: 
                 
                testdata/logic_test/role:400: SELECT * FROM information_schema.applicable_roles
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolec      NO
                    testuser  roleb      NO
                    testuser  rolea      YES
                    testuser  roled      NO
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2206: 
                testdata/logic_test/role:409: error while processing
            logic.go:2206: testdata/logic_test/role:409: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/022107715d0592d305e3244f6f47e913f0942479

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1595976&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.42s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic855745178
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
--- done: testdata/logic_test/restore with config local: 1 tests, 0 failures
rewrote:
CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));


        --- FAIL: TestCCLLogic/local/role (1.43s)
            logic.go:1662: 
                 
                testdata/logic_test/role:357: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2206: 
                testdata/logic_test/role:363: error while processing
            logic.go:2206: testdata/logic_test/role:363: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/1c267f3b894abcc9b10dfcf475e9a0f3146a8de4

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1598037&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (7.17s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic159827376
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
rewrote:
CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));


--- done: testdata/logic_test/case_sensitive_names with config local: 2 tests, 0 failures
rewrote:
CREATE TABLE orders (region STRING NOT NULL, id INT8 NOT NULL, total DECIMAL NOT NULL, created_at TIMESTAMP NOT NULL, PRIMARY KEY (region, id), FAMILY (total, created_at));


        --- FAIL: TestCCLLogic/local/role (1.53s)
            logic.go:1662: 
                 
                testdata/logic_test/role:394: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2206: 
                testdata/logic_test/role:400: error while processing
            logic.go:2206: testdata/logic_test/role:400: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/1c3fce73cd25fa69e5a8c05ce8e215d66b4f49e4

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1599900&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.90s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic049992869
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.14s)
            logic.go:1652: 
                 
                testdata/logic_test/role:198: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  testrole   YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:204: error while processing
            logic.go:2196: testdata/logic_test/role:204: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/205330426e50b966ede8a4644d77570d8265fe66

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1602204&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.52s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic247588810
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.78s)
            logic.go:1652: 
                 
                testdata/logic_test/role:542: SELECT * FROM information_schema.applicable_roles
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    testuser  roleb      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:549: error while processing
            logic.go:2196: testdata/logic_test/role:549: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/85b76090c30a58e43f778818795783eaca3758ff

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1605106&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (7.24s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic362822206
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/crdb_internal/privileged_zones_test
--- done: testdata/logic_test/crdb_internal with config local: 10 tests, 2 failures
--- done: testdata/logic_test/role with config local: 156 tests, 0 failures
            --- FAIL: TestCCLLogic/local/crdb_internal/privileged_zones_test (0.38s)
                logic.go:1652: 
                     
                    testdata/logic_test/crdb_internal:113: SHOW ZONE CONFIGURATION FOR DATABASE db2
                    expected "pq: user testuser has no privileges on database db2", but no error occurred
                logic.go:1304: 
                     testdata/logic_test/crdb_internal:116: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local/crdb_internal
=== PAUSE TestCCLLogic/local/crdb_internal
=== CONT  TestCCLLogic/local/crdb_internal
--- done: testdata/logic_test/case_sensitive_names with config local: 2 tests, 0 failures
--- total progress: 49 statements/queries
--- done: testdata/logic_test/partitioning with config local: 138 tests, 0 failures
rewrote:
CREATE TABLE t1 (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c), INDEX (a, b) PARTITION BY LIST (a, b) (PARTITION p00 VALUES IN ((0, 0)))) PARTITION BY LIST (a) (PARTITION p12 VALUES IN (1, 2) PARTITION BY LIST (b) (PARTITION p12p3 VALUES IN (3) PARTITION BY LIST (c) (PARTITION p12p3p8 VALUES IN (8)), PARTITION pd VALUES IN (DEFAULT)), PARTITION p6 VALUES IN (6) PARTITION BY RANGE (b) (PARTITION p6p7 VALUES FROM (minvalue) TO (7), PARTITION p6p8 VALUES FROM (7) TO (8), PARTITION p6px VALUES FROM (8) TO (maxvalue)));


        --- FAIL: TestCCLLogic/local/crdb_internal (0.79s)
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/1179c76e34d057259854d88c7948830d2b4b05cd

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1607022&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (7.19s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic975987550
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
rewrote:
CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));


--- done: testdata/logic_test/as_of with config local: 3 tests, 0 failures
rewrote:
CREATE TABLE orders (region STRING NOT NULL, id INT8 NOT NULL, total DECIMAL NOT NULL, created_at TIMESTAMP NOT NULL, PRIMARY KEY (region, id), FAMILY (total), FAMILY (created_at));


--- done: testdata/logic_test/restore with config local: 1 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (2.20s)
            logic.go:1652: 
                 
                testdata/logic_test/role:394: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:400: error while processing
            logic.go:2196: testdata/logic_test/role:400: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/4de1b957fbda6eaa42c8fd7b0fb3278365f7a615

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1607807&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.15s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic911289440
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.61s)
            logic.go:1652: 
                 
                testdata/logic_test/role:357: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:363: error while processing
            logic.go:2196: testdata/logic_test/role:363: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/60b7b3b2ca511cf9adb30799fcf2b8cffd6223d9

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1608678&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (7.10s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic490882357
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
rewrote:
CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));


--- done: testdata/logic_test/case_sensitive_names with config local: 2 tests, 0 failures
rewrote:
CREATE TABLE orders (region STRING NOT NULL, id INT8 NOT NULL, total DECIMAL NOT NULL, created_at TIMESTAMP NOT NULL, PRIMARY KEY (region, id), FAMILY (created_at), FAMILY (total));


        --- FAIL: TestCCLLogic/local/role (1.67s)
            logic.go:1652: 
                 
                testdata/logic_test/role:409: SELECT * FROM information_schema.enabled_roles
                expected:
                    role_name
                    rolea
                    roleb
                    rolec
                    roled
                    testuser
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    role_name
                    testuser
                    
                
            logic.go:2196: 
                testdata/logic_test/role:419: error while processing
            logic.go:2196: testdata/logic_test/role:419: too many errors encountered, skipping the rest of the input

```

SHA: https://github.com/cockroachdb/cockroach/commits/78f8482fd864376a26f32dd983afeb679d1c5977

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1611255&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (7.10s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic239458900
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (2.01s)
            logic.go:1652: 
                 
                testdata/logic_test/role:406: SELECT * FROM information_schema.applicable_roles
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolec      NO
                    testuser  roleb      NO
                    testuser  rolea      YES
                    testuser  roled      NO
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:415: error while processing
            logic.go:2196: testdata/logic_test/role:415: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/f97dc13163020a032b098ef3eb88e4d9f54a04ba

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1614388&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.44s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic644457114
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (0.87s)
            logic.go:1652: 
                 
                testdata/logic_test/role:204: SELECT * FROM information_schema.applicable_roles
                expected:
                    grantee   role_name  is_grantable
                    testuser  testrole   YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:210: error while processing
            logic.go:2196: testdata/logic_test/role:210: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/a60c4680e82b390ab058634a58626f56c80b27ab

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1617014&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.74s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic048581627
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
rewrote:
CREATE TABLE abc (a INT8, b INT8, c INT8, PRIMARY KEY (a, b, c)) PARTITION BY LIST (a, b) (PARTITION small VALUES IN ((1, 1), (2, 2), (3, 3)));


rewrote:
CREATE TABLE orders (region STRING NOT NULL, id INT8 NOT NULL, total DECIMAL NOT NULL, created_at TIMESTAMP NOT NULL, PRIMARY KEY (region, id), FAMILY (total), FAMILY (created_at));


--- done: testdata/logic_test/restore with config local: 1 tests, 0 failures
--- done: testdata/logic_test/as_of with config local: 3 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (1.77s)
            logic.go:1652: 
                 
                testdata/logic_test/role:378: SELECT * FROM information_schema.enabled_roles
                expected:
                    role_name
                    rolea
                    roleb
                    rolec
                    roled
                    testuser
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    role_name
                    testuser
                    
                
            logic.go:2196: 
                testdata/logic_test/role:388: error while processing
            logic.go:2196: testdata/logic_test/role:388: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/581a2bff2f50d5626f68bc1d51af0cf05b455dfc

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1618973&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.82s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic122010987
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.70s)
            logic.go:1652: 
                 
                testdata/logic_test/role:400: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:406: error while processing
            logic.go:2196: testdata/logic_test/role:406: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/6c13f01ef0d999095a16345b21fc455648796e0c

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1619864&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (5.26s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic444818117
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.05s)
            logic.go:1652: 
                 
                testdata/logic_test/role:378: SELECT * FROM information_schema.enabled_roles
                expected:
                    role_name
                    rolea
                    roleb
                    rolec
                    roled
                    testuser
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    role_name
                    testuser
                    
                
            logic.go:2196: 
                testdata/logic_test/role:388: error while processing
            logic.go:2196: testdata/logic_test/role:388: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/d3574ad671bd3631e780510235485681720c2b8f

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1622141&tab=buildLog

```


rewrote:
CREATE TABLE t (a INT8, b INT8, PRIMARY KEY (a, b)) PARTITION BY RANGE (a, b) (PARTITION p1 VALUES FROM (maxvalue, minvalue) TO (maxvalue, maxvalue));


rewrote:
CREATE TABLE t (a INT8, b INT8, PRIMARY KEY (a, b)) PARTITION BY RANGE (a, b) (PARTITION p1 VALUES FROM (minvalue, maxvalue) TO (maxvalue, maxvalue));


rewrote:
CREATE TABLE solar_system (planet STRING, region STRING, subregion STRING, val INT8, PRIMARY KEY (planet, region, subregion, val)) PARTITION BY LIST (planet, region, subregion) (PARTITION westcoast VALUES IN (('earth', 'us', 'seatle'), ('earth', 'us', 'cali')), PARTITION eu VALUES IN (('earth', 'eu', DEFAULT)), PARTITION us VALUES IN (('earth', 'us', DEFAULT)), PARTITION earth VALUES IN (('earth', DEFAULT, DEFAULT)), PARTITION mars VALUES IN (('mars', DEFAULT, DEFAULT)), PARTITION jupiter VALUES IN (('jupiter', DEFAULT, DEFAULT)), PARTITION titan VALUES IN (('jupiter', 'titan', DEFAULT)), PARTITION red_spot VALUES IN (('jupiter', 'titan', 'red spot')));


rewrote:
CREATE TABLE foo (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (1.2));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (0) PARTITION BY LIST (b) (PARTITION p1_1 VALUES IN (1.2)));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (minvalue));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a, b) (PARTITION p1 VALUES IN ((1, 1), (1, (minvalue))));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN (maxvalue));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a, b) (PARTITION p1 VALUES IN ((1, 1), (1, (maxvalue))));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a) (PARTITION p1 VALUES FROM (DEFAULT) TO (0));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a) (PARTITION p1 VALUES FROM (0) TO (DEFAULT));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a) (PARTITION p1 VALUES FROM ((DEFAULT)) TO (0));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a) (PARTITION p1 VALUES FROM (0) TO ((DEFAULT)));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a, b) (PARTITION p1 VALUES FROM (0, (DEFAULT)) TO (0, 0));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY RANGE (a, b) (PARTITION p1 VALUES FROM (0, 0) TO (0, (DEFAULT)));


rewrote:
CREATE TABLE t (a INT8, b INT8, c INT8, PRIMARY KEY (a, b), FAMILY (c)) PARTITION BY LIST (a) (PARTITION p1 VALUES IN ((SELECT 1)));


--- done: testdata/logic_test/partitioning_constrained_scans with config local: 25 tests, 0 failures
        --- FAIL: TestCCLLogic/local/role (2.10s)
            logic.go:1652: 
                 
                testdata/logic_test/role:406: SELECT * FROM information_schema.applicable_roles
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolec      NO
                    testuser  roleb      NO
                    testuser  rolea      YES
                    testuser  roled      NO
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:415: error while processing
            logic.go:2196: testdata/logic_test/role:415: too many errors encountered, skipping the rest of the input
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/ed717cbaf741e3a32c76db25b16a59dc2a8221d7

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1624180&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.99s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic238865283
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.95s)
            logic.go:1652: 
                 
                testdata/logic_test/role:400: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:406: error while processing
            logic.go:2196: testdata/logic_test/role:406: too many errors encountered, skipping the rest of the input

```

SHA: https://github.com/cockroachdb/cockroach/commits/e1c1a9ae765730782f3adc2298a76d6a07ad055d

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestCCLLogic PKG=github.com/cockroachdb/cockroach/pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1626731&tab=buildLog

```
=== RUN   TestCCLLogic
--- FAIL: TestCCLLogic (6.64s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestCCLLogic119602870
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/role
=== PAUSE TestCCLLogic/local/role
=== CONT  TestCCLLogic/local/role
        --- FAIL: TestCCLLogic/local/role (1.86s)
            logic.go:1652: 
                 
                testdata/logic_test/role:541: SELECT * FROM information_schema.administrable_role_authorizations
                expected:
                    grantee   role_name  is_grantable
                    testuser  rolea      YES
                    testuser  roleb      YES
                    
                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable
                    
                
            logic.go:2196: 
                testdata/logic_test/role:548: error while processing
            logic.go:2196: testdata/logic_test/role:548: too many errors encountered, skipping the rest of the input

```

Meant to send this earlier. Here is how to reproduce the test failure:

```
~ make stress PKG=./ccl/logictestccl/ TESTS='TestCCLLogic/local/(role|crdb_internal)' TESTFLAGS=-v
...
    --- FAIL: TestCCLLogic/local (0.00s)
        --- FAIL: TestCCLLogic/local/role (0.94s)
            logic.go:1652:

                testdata/logic_test/role:204: SELECT * FROM information_schema.applicable_roles
                expected:
                    grantee   role_name  is_grantable
                    testuser  testrole   YES

                but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
                    grantee  role_name  is_grantable


            logic.go:2196:
                testdata/logic_test/role:210: error while processing
            logic.go:2196: testdata/logic_test/role:210: too many errors encountered, skipping the rest of the input
        --- PASS: TestCCLLogic/local/crdb_internal (0.97s)
            --- PASS: TestCCLLogic/local/crdb_internal/privileged_zones_test (0.62s)
```

The problematic interaction is that `crdb_internal` does a `GRANT` concurrently with the running of the `role` logic test. Again this points to the `roleMembersCache` global being the problem.

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -702,6 +702,7 @@ func NewServer(cfg Config, stopper *stop.Stopper) (*Server, error) {
 		HistogramWindowInterval: s.cfg.HistogramWindowInterval(),
 		RangeDescriptorCache:    s.distSender.RangeDescriptorCache(),
 		LeaseHolderCache:        s.distSender.LeaseHolderCache(),
+		RoleMemberCache:         &sql.MembershipCache{},
 		TestingKnobs:            sqlExecutorTestingKnobs,
 
 		DistSQLPlanner: sql.NewDistSQLPlanner(

--- a/pkg/sql/authorization.go
+++ b/pkg/sql/authorization.go
@@ -22,14 +22,13 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 )
 
-type membershipCache struct {
+// MembershipCache is a shared cache for role membership information.
+type MembershipCache struct {
 	syncutil.Mutex
 	tableVersion sqlbase.DescriptorVersion
 	// userCache is a mapping from username to userRoleMembership.
 	userCache map[string]userRoleMembership
 }
-
-var roleMembersCache membershipCache
 
 // userRoleMembership is a mapping of "rolename" -> "with admin option".
 type userRoleMembership map[string]bool
@@ -195,6 +194,8 @@ func (p *planner) RequireAdminRole(ctx context.Context, action string) error {
 func (p *planner) MemberOfWithAdminOption(
 	ctx context.Context, member string,
 ) (map[string]bool, error) {
+	roleMembersCache := p.execCfg.RoleMemberCache
+
 	// Lookup table version.
 	objDesc, err := p.PhysicalSchemaAccessor().GetObjectDesc(ctx, p.txn, &roleMembersTableName,
 		p.ObjectLookupFlags(true /*required*/, false /*requireMutable*/))
@@ -212,7 +213,7 @@ func (p *planner) MemberOfWithAdminOption(
 		roleMembersCache.Lock()
 		if roleMembersCache.tableVersion != tableVersion {
 			// Update version and drop the map.
-			roleMembersCache.tableVersion = tableDesc.Version
+			roleMembersCache.tableVersion = tableVersion
 			roleMembersCache.userCache = make(map[string]userRoleMembership)
 		}
 
@@ -242,7 +243,6 @@ func (p *planner) MemberOfWithAdminOption(
 		// Table version remains the same: update map, unlock, return.
 		roleMembersCache.userCache[member] = memberships
 		roleMembersCache.Unlock()
-
 		return memberships, nil
 	}
 }

--- a/pkg/sql/exec_util.go
+++ b/pkg/sql/exec_util.go
@@ -530,6 +530,9 @@ type ExecutorConfig struct {
 	// Caches updated by DistSQL.
 	RangeDescriptorCache *kv.RangeDescriptorCache
 	LeaseHolderCache     *kv.LeaseHolderCache
+
+	// Role membership cache.
+	RoleMemberCache *MembershipCache
 }
 
 // Organization returns the value of cluster.organization.

